core/src/com/unciv/ui/components/widgets/HealthBarV0.kt

```diff
@@ -0,0 +1,317 @@
+package com.unciv.ui.components.widgets
+
+import com.badlogic.gdx.graphics.Color
+import com.badlogic.gdx.graphics.g2d.Batch
+import com.badlogic.gdx.math.Interpolation
+import com.badlogic.gdx.math.Rectangle
+import com.badlogic.gdx.scenes.scene2d.actions.FloatAction
+import com.badlogic.gdx.scenes.scene2d.ui.Label
+import com.badlogic.gdx.scenes.scene2d.ui.Widget
+import com.badlogic.gdx.scenes.scene2d.utils.Drawable
+import com.badlogic.gdx.utils.Align
+import com.unciv.logic.city.City
+import com.unciv.models.ruleset.INonPerpetualConstruction
+import com.unciv.ui.components.extensions.darken
+import com.unciv.ui.components.fonts.Fonts
+import com.unciv.ui.components.tilegroups.CityButton
+import com.unciv.ui.images.ImageGetter
+import com.unciv.ui.screens.basescreen.BaseScreen
+import kotlin.math.PI
+import kotlin.math.cos
+
+@Suppress("MemberVisibilityCanBePrivate", "unused")
+
+//TODO Since HealthBar instances can share style instances with the same values, use common ones.
+//      - skin, dedicated cache, enum, or companion vals?
+
+/**
+ *  A reusable Widget to draw progress bars, health bars and the like.
+ */
+
+class HealthBarV0(
+    val minValue: Float,
+    val maxValue: Float,
+    segmentCount: Int = 2,
+    isVertical: Boolean = false,
+    var style: HealthBarStyle = HealthBarStyle(isVertical)
+) : Widget() {
+    /**
+     *  A reusable Widget to draw progress bars, health bars and the like.
+     *  Convenience constructor using Int limits, and default constructor.
+     *
+     *  Note that a default instance without any further configuration will draw nothing and have zero pref size.
+     */
+    constructor(
+        minValue: Int = 0,
+        maxValue: Int = 100,
+        segmentCount: Int = 2,
+        isVertical: Boolean = false
+    ) : this(minValue.toFloat(), maxValue.toFloat(), segmentCount, isVertical)
+
+    class HealthBarStyle(var isVertical: Boolean) {
+        /** The background, if any, also defines the actual bar area by its leftWidth, topHeight etc properties */
+        var background: Drawable? = null
+        /** In case a background fails to define leftWidth etc we can reduce the bar area with this */
+        var pad: Float
+            get() = (padHorizontal + padVertical) / 2
+            set(value) { padHorizontal = value; padVertical = value }
+        /** @see pad */
+        var padVertical: Float = 0f
+        /** @see pad */
+        var padHorizontal: Float = 0f
+
+        /** At layout time, there must be enough colors. One per segment, except the last one can be omitted and will mean Color.CLEAR */
+        var colors: Array<Color> = arrayOf(Color.GREEN)
+
+        var minBarWidth: Float = 0f
+        var minBarHeight: Float = 0f
+
+        var animateDuration: Float = 0f
```

Comment:
durationSeconds please - I hate having to guess seconds or milliseconds 
Same for the rest